### ISSUE TITLE
[vm/ffi] Pointer<T extends Struct> should expose an Iterable

### ISSUE THREAD

**Issue by safasofuoglu** [dart-lang/sdk#46560]
Having a struct definition, a pointer and an item count, it felt convenient to work with an iterator instead of elementAt and ref calls. I started with:

```dart
class NativeArray<T extends Struct> implements UnmodifiableListView<T> {
  Pointer<T> pointer;

  NativeArray(this.pointer, this.length);

  @override
  T get first {
    return pointer.elementAt(0);
  }
  ...
}
```

However, for the elementAt call, the compiler says T must be a compile-time constant. 

Can this be made possible on userland? If not, would you consider implementing it on the sdk?

@dcharkes

**Comment by dcharkes** [comment]
> Can this be made possible on userland? If not, would you consider implementing it on the sdk?

No, we disallow unbound type arguments (see the documentation of `.ref`) for performance reasons.

On `Pointer`s to integers one can use `asTypedList` to get an iterator. But on `Pointer<T extends Struct>` we have nothing yet.

(Cross reference other missing iterator in the API: https://github.com/dart-lang/sdk/issues/45508)

**Comment by safasofuoglu** [comment]
Thanks @dcharkes , #45508 looks good.

**Comment by dcharkes** [comment]
Well, we still need to implement this.
